File: InternetBulletin.Shared/DTOs/Posts/AddPostDTO.cs
// *********************************************************************************
//	<copyright file="AddPostDTO.cs" company="Personal">
//		Copyright (c) 2025 Personal
//	</copyright>
// <summary>Add post dto.</summary>
// *********************************************************************************

namespace InternetBulletin.Shared.DTOs.Posts
{
	/// <summary>
	/// Add post dto.
	/// </summary>
	public class AddPostDTO
	{
		/// <summary>
		/// Gets or sets the post title.
		/// </summary>
		/// <value>
		/// The post title.
		/// </value>
		public string PostTitle { get; set; } = string.Empty;

		/// <summary>
		/// Gets or sets the content of the post.
		/// </summary>
		/// <value>
		/// The content of the post.
		/// </value>
		public string PostContent { get; set; } = string.Empty;

		/// <summary>
		/// Gets or sets the post created by.
		/// </summary>
		/// <value>
		/// The post created by.
		/// </value>
		public string PostCreatedBy { get; set; } = string.Empty;

		/// <summary>
		/// Gets or sets a value indicating whether this instance is NSFW.
		/// </summary>
		/// <value>
		///   <c>true</c> if this instance is NSFW; otherwise, <c>false</c>.
		/// </value>
		public bool IsNSFW { get; set; }

		/// <summary>
		/// Gets or sets the genre tag.
		/// </summary>
		/// <value>
		/// The genre tag.
		/// </value>
		public string GenreTag { get; set; } = string.Empty;

	}
}

File: IBBS.UI/Models/AddPostDto.js
/**
 * AddPostDtoModel class to represent the data for adding a post.
 */
class AddPostDtoModel {
	/**
	 * Gets or sets the post title.
	 * @type {string}
	 */
	postTitle = "";

	/**
	 * Gets or sets the content of the post.
	 * @type {string}
	 */
	postContent = "";

	/**
	 * Gets or sets the post created by.
	 * @type {string}
	 */
	postCreatedBy = "";

	/**
	 * Gets or sets the IsNSFW flag.
	 * @type {boolean}
	 */
	isNSFW = false;

	/**
	 * Gets or sets the genre tag.
	 * @type {string}
	 */
	genreTag = "";

	/**
	 * Creates an instance of AddPostDtoModel.
	 * @param {string} PostTitle The title of the post.
	 * @param {string} PostContent The content of the post.
	 * @param {string} PostCreatedBy The user who created the post.
	 * @param {boolean} IsNSFW The NSFW rating.
	 * @param {string} GenreTag The genre tag.
	 */
	constructor(PostTitle, PostContent, PostCreatedBy, IsNSFW, GenreTag) {
		this.postTitle = PostTitle;
		this.postContent = PostContent;
		this.postCreatedBy = PostCreatedBy;
		this.isNSFW = IsNSFW;
		this.genreTag = GenreTag;
	}
}

export default AddPostDtoModel;
